### PR TITLE
Fix stuck ACs

### PR DESCRIPTION
**Pull request by Atlante45**
This PR should fix the issue where a large number of assignment clients are left behind after the server shutdown.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4927/HighFidelity-Beta-PR11810-b31118fe002569e8e1df9d376d9a9af8d35ec17d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4927/HighFidelity-Beta-PR11810-b31118fe002569e8e1df9d376d9a9af8d35ec17d.dmg

**Comment by Atlante45** [comment]
[Test Plan](https://highfidelity.testrail.net/index.php?/cases/view/2969)

File: libraries/shared/src/SharedUtil.cpp [hunk 1]
```diff
@@ -1113,3 +1113,56 @@ void watchParentProcess(int parentPID) {
...
+#ifdef Q_OS_WIN
+QString getLastErrorAsString() {
+    DWORD errorMessageID = ::GetLastError();
+    if (errorMessageID == 0) {
+        return QString();
+    }
+
+    LPSTR messageBuffer = nullptr;
+    size_t size = FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+        NULL, errorMessageID, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPSTR)&messageBuffer, 0, NULL);
```
Comment (huffman) [inline libraries/shared/src/SharedUtil.cpp]: Can these be `nullptr` instead of `NULL`?

File: libraries/shared/src/SharedUtil.cpp [hunk 2]
```diff
@@ -1113,3 +1113,56 @@ void watchParentProcess(int parentPID) {
...
+
+    auto message = QString::fromLocal8Bit(messageBuffer, (int)size);
+
+    //Free the buffer.
+    LocalFree(messageBuffer);
+
+    return message;
+}
+
+void* createJobObject() {
```
Comment (huffman) [inline libraries/shared/src/SharedUtil.cpp]: The only comment I'd make here is to maybe make it clearer that this job object will kill attached processes.

**Comment by cain-kilgore** [comment]
Testing.

File: libraries/shared/src/SharedUtil.cpp (hunk 1 above)
Comment (Atlante45) [reply libraries/shared/src/SharedUtil.cpp]: I wasn't sure, but I dug a bit into it and it looks like that's fine. Will fix.

**Comment by cain-kilgore** [comment]
# Test Failed
[TestRail Link](https://highfidelity.testrail.net/index.php?/tests/view/13323&group_by=cases:section_id&group_order=asc&group_id=624)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4945/HighFidelity-Beta-PR11810-de1ec0c90b28fa960445bdfee8a12999ba033b24.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4945/HighFidelity-Beta-PR11810-de1ec0c90b28fa960445bdfee8a12999ba033b24.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4952/HighFidelity-Beta-PR11810-d8b1cb10dfb9a785128e6c1339a7362a89f98a10.exe

**Comment by Atlante45** [comment]
@cain-kilgore Can you give it another test?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4952/HighFidelity-Beta-PR11810-d8b1cb10dfb9a785128e6c1339a7362a89f98a10.dmg

**Comment by cain-kilgore** [comment]
@Atlante45: Can do, testing.

**Comment by cain-kilgore** [comment]
# Test Passed
[TestRail Link](https://highfidelity.testrail.net/index.php?/tests/view/13323&group_by=cases:section_id&group_order=asc&group_id=624)

**Pull request merged**